### ISSUE TITLE
Keep aspect ratio of images?

### ISSUE THREAD

**Issue by javiabellan** [Parskatt/DKM#62]
I noticed that DKM resizes the image to a fixed size and aspect ratio (horizontal). I think preserving the original aspect ratio of the images could improve performance, especially for extreme image pairs like this one:

| | | 
|-|-|
| ![BARCELONA -VUNAV-1024x682 2](https://github.com/Parskatt/DKM/assets/31625296/32d0db21-1021-47db-896d-5fb36787cb9d) | ![BARCELONA -VUNAV-1024x682 3](https://github.com/Parskatt/DKM/assets/31625296/5d5db361-dc5c-4f2d-b3ab-9f6207cdad4c) |

There is any tweak (code modification) to allow this into the current DKM weights?

**Comment by Parskatt** [comment]
Yes it possible. And for the pair you show I agree it would probably improve performance.

If you want to dig into the code, the main barriers are

1. For feature extraction the images are stacked. Instead we need to extract seperately.

2. In several places I just use "H,W". This needs to be H_A/H_B or something.

In tiny roma (over in the roma repo), I've made these changes. You can refer to the implementation there.
